### PR TITLE
Externalizing BloodCat placements

### PR DESCRIPTION
**Pull request by ja2-derek**
Externalizing strategic map data from code to json. Bloodcat placements are relatively simple and independent so I am starting here.

See #665 and #1011. 

## Summary

- There are 2 types of placements. First is the "regular" placement which are slots for bloodcat ambush. Second are bloodcats that are always present in the Lair and Arena sectors, which I call "spawns"
- Added models classes and deserialization for Bloodcat placements
- Updated GameInit to use externalized data
- Replaced a few places that uses `SEC_I16` and `SEC_N5` with the externalized fact `isLair` and `isArena`  

## LoadSave

There are some BloodCat placement patching logic in [LoadStrategicAI](https://github.com/ja2-stracciatella/ja2-stracciatella/blob/9f76684de4cbab33ad8e31d756d2e6f5de44ede5/src/game/Strategic/Strategic_AI.cc#L2633), for `( ubSAIVersion < 18 )` and `f( ubSAIVersion < 22 )`.

I suppose these are for only patching saves from very old JA2 versions. I am not touching those.

### CODE REVIEW

**Comment by lynxlynxlynx** [comment]
LGTM, thanks! How much testing did this get?

**Comment by ja2-derek** [comment]
I am trying this in-game, launched the debugger and eye-balled the deserialized values. GABBI-teleported to the Lair. Any better way to test it? 

Also is the structure and style OK? This is my first major PR, will continue doing similar things for other externalization.

Probably this and the next few PRs can wait after 0.17 release.

**Comment by lynxlynxlynx** [comment]
That's fine and I don't have any architectural comments either. I'll leave it open for a bit if anyone else wants to review.

File: src/game/Strategic/Campaign_Types.h [hunk 1]
```diff
@@ -16,6 +16,15 @@ static inline UINT8 SECTOR(UINT8 const x, UINT8 const y)
 	return (y - 1) * 16 + x - 1;
 }
 
+//Macro to convert sector IDs such as A1,P16 to 0-255
+static inline UINT8 SECTOR_FROM_COORDINATES(const char* coordinates)
```
Comment (ja2-derek) [inline src/game/Strategic/Campaign_Types.h:20]: Realized this should be called "Short Sector String". (See StrategicMap.cc:GetShortSectorString))

**Comment by lynxlynxlynx** [comment]
Thanks!

**Pull request merged**